### PR TITLE
ffmpeg: fix cross-build (at least for Apple OS)

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

closes https://github.com/conan-io/conan-center-index/issues/7256

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by MartinDelille — DISMISSED** [review]

File: recipes/ffmpeg/all/conanfile.py [hunk 1]
```diff
@@ -108,23 +108,19 @@ def config_options(self):
             del self.options.with_pulse
         if self.settings.os != "Macos":
             del self.options.with_appkit
-            del self.options.with_avfoundation
+        if self.settings.os not in ["Macos", "iOS", "tvOS"]:
```
Comment (madebr) [inline recipes/ffmpeg/all/conanfile.py:111]: Would `tools.is_apple_os` do the job too?
```suggestion
        if not tools.is_apple_os(self.settings.os):
```

File: recipes/ffmpeg/all/conanfile.py [hunk 2]
```diff
@@ -176,24 +172,32 @@ def requirements(self):
     def validate(self):
         if self.options.with_ssl == "securetransport" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("securetransport is only available on Apple")
-        if self.settings.os == "Macos" and self.settings.arch == "armv8" and self.options.with_libvpx:
-            raise ConanInvalidConfiguration("libvpx doesn't support armv8 (yet)")
-        if self.settings.os in ["Linux", "FreeBSD"] and self.options.with_sdl:
-            raise ConanInvalidConfiguration("sdl is not supported by this recipe (yet)")
+
+    def build_requirements(self):
+        self.build_requires("yasm/1.3.0")
```
Comment (madebr) [inline recipes/ffmpeg/all/conanfile.py:177]: I
```suggestion
        if self.settings.arch in ("x86", "x86_64"):
            self.build_requires("yasm/1.3.0")
```

Comment (SSE4) [reply recipes/ffmpeg/all/conanfile.py:177]: what is the warning? isn't yaml capable to generate ASM other from x86/x86_64?

Comment (madebr) [reply recipes/ffmpeg/all/conanfile.py:177]: *yasm*  :wink: 
I don't think so.
[Their website](https://yasm.tortall.net/) only mentions *x86 and AMD64 instruction sets*

Comment (SSE4) [reply recipes/ffmpeg/all/conanfile.py:177]: yeah, and I remember [ffmpeg builds instructions](https://trac.ffmpeg.org/wiki/CompilationGuide/WinRT) mentioned to use [gas-preprocessor](https://github.com/libav/gas-preprocessor) for the ARM (maybe few other architectures). I assume it's not yet packaged in ConanCenter.
so probably, we have to build with `--disable-asm` for other architectures besides x86/x86_64 for a while.

File: recipes/ffmpeg/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/ffmpeg/all/conanfile.py:111]: No, because watchOS must be excluded here

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**